packages/core/src/Synchronizer.ts

```diff
@@ -297,16 +335,21 @@ export class Synchronizer extends EventEmitter {
 
     private async _poll(): Promise<{ complete: boolean }> {
         if (!this.setupComplete) {
-            console.warn('polled before setup, nooping')
+            console.warn(
+                '@unirep/core:Synchronizer: polled before setup, nooping'
+            )
             return { complete: false }
         }
         this.emit('pollStart')
+
         const state = await this._db.findOne('SynchronizerState', {
```

Comment:
the test breaks because it cannot find the object (`{ where: attesterId }`) in the same db path (`:memory:`)
temporarily solved this by using `upsert` and `db.transaction` [d7049d1](https://github.com/Unirep/Unirep/pull/367/commits/d7049d10cdf68fa0e46f78ad1609877f59376f07)


> Say i am syncing attester `a`, `b`, and `c`. I synchronize to block 30, then stop the synchronizer and add attester `d`. I restart the synchronizer.
> 
> Now events from blocks 0-30 will be double processed for `a`, `b`, and `c`.

I imagine that we fix the attester IDs so that in this PR it doesn't solve dynamic attester IDs.
Or it will be like `add(attesterID)`, `remove(attesterID)`

Reply:
No the test failed because `findStartBlock` attempts to create a `SynchronizerState` with an `attesterId` that already exists. There's still a problem where something like `handleUserSignedUp` is creating duplicate entries in `UserSignUp`.

This happens when a database is re-used. Each handler should check if the object exists before trying to create the object.

Reply:
This problem happens with all handlers when the `_attesterId` array is changed and the synchronizer is restarted with the same database.

Reply:
Yeah I mean this errors in [d3ca02b](https://github.com/Unirep/Unirep/pull/367/commits/d3ca02b703293465d14bbb9274674cbdb3b98f53)
<img width="975" alt="" src="https://user-images.githubusercontent.com/17507085/230699948-d26a71dc-cca2-442f-9bb9-556ada0181fb.png">

and solved by [d7049d1](https://github.com/Unirep/Unirep/pull/367/commits/d7049d10cdf68fa0e46f78ad1609877f59376f07)

Reply:
Yes but 

> the test breaks because it cannot find the object ({ where: attesterId })

is not correct, the problem is the object already exists and so is being duplicated because we're re-using the `db` as a part of the test [here](https://github.com/Unirep/Unirep/pull/367/commits/d3ca02b703293465d14bbb9274674cbdb3b98f53#diff-bc918440d2f3a5bd3dd0142bdec0b7804e96f455b9b2ea43a022f0b8962657eaR258). The test should technically still fail because there are duplicate entries for other collections like `UserSignUp`, i'll update the test

Reply:
> This problem happens with all handlers when the `_attesterId` array is changed and the synchronizer is restarted with the same database.

yeah I mean this PR doesn't assume `_attesterId` be dynamic
so the event will be processed sequentially 
and if it is `_syncAll` and it will catch the attester ID from the blocknumber the attester signs up

Reply:
Ah [d7049d1](https://github.com/Unirep/Unirep/pull/367/commits/d7049d10cdf68fa0e46f78ad1609877f59376f07) is a good fix, i was causing a race condition by calling `start` twice